src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = new User();
+        user.setLogin("Good login");
+        user.setPassword("Good password");
+        user.setAge(19);
+
+        registrationService.register(user);
+
+        User newUser = new User();
+        newUser.setLogin("Good login");
+        newUser.setPassword("Good password");
+        newUser.setAge(19);
+
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(newUser),
+                "Expected register() to throw, but it didn't");
+    }
+
+    @Test
+    public void register_shortLogin_notOK() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("Good password");
+        user.setAge(19);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));

```

Comment:
The test `register_shortLogin_notOK` does not define what is considered a 'short' login. According to the checklist, edge cases should be tested, so the exact length that is considered too short should be included in the test.

Reply:
"short" is 5 characters length. It is edge case